src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/X86Base.cs

```diff
@@ -62,5 +67,19 @@ internal abstract class X64
         /// Its functionality is exposed in the public <see cref="System.Numerics.BitOperations" /> class.
         /// </remarks>
         internal static uint BitScanReverse(uint value) => BitScanReverse(value);
+
+        /// <summary>
+        /// void __cpuidex(int cpuInfo[4], int function_id, int subfunction_id);
+        ///   CPUID
+        /// </summary>
+        public static unsafe (int Eax, int Ebx, int Ecx, int Edx) CpuId(int functionId, int subFunctionId)
+        {
+            int* cpuInfo = stackalloc int[4];
+            __cpuidex(cpuInfo, functionId, subFunctionId);
+            return (cpuInfo[0], cpuInfo[1], cpuInfo[2], cpuInfo[3]);
+        }
+
+        [DllImport(RuntimeHelpers.QCall)]
```

Comment:
This will need implementation in Mono too.

Reply:
Is there any documentation on what will be needed for Mono?

Reply:
> Is there any documentation on what will be needed for Mono?

@marek-safar @vargaz Could you please help to route this?

Reply:
I guess we'll implement that as intrinsic as we have access to llvm version Anyway, this section needs to move to vm specific file and can throw PNSE for now.

/cc @SamMonoRT to track this further

Reply:
Is there a msbuild property I can use to have the `PlatformNotSupported.cs` version of the file used on Mono?

Reply:
You need to extract only this single method, follow https://github.com/dotnet/runtime/tree/master/src/libraries/System.Private.CoreLib/src#systemprivatecorelib-shared-sources doc